crates/turborepo-lib/src/commands/logout.rs

```diff
@@ -1,15 +1,28 @@
-use anyhow::Result;
-use tracing::error;
+use anyhow::{anyhow, Result};
 use turborepo_ui::GREY;
 
-use crate::commands::CommandBase;
+use crate::{commands::CommandBase, rewrite_json::unset_path};
 
 pub fn logout(base: &mut CommandBase) -> Result<()> {
-    if let Err(err) = base.user_config_mut()?.set_token(None) {
-        error!("could not logout. Something went wrong: {}", err);
-        return Err(err.into());
-    }
+    let global_config_path = base.global_config_path()?;
+    let before = global_config_path.read_to_string().or_else(|e| {
+        if matches!(e.kind(), std::io::ErrorKind::NotFound) {
+            Ok(String::from("{}"))
+        } else {
+            Err(anyhow!(
+                "Encountered an IO error while attempting to read {}: {}",
+                global_config_path,
+                e
+            ))
+        }
+    })?;
+    let output = if let Some(after) = unset_path(&before, &["token"], true)? {
+        global_config_path.create_with_contents(after)?;
+        ">>> Logged out"
+    } else {
+        ">>> Not logged in"
+    };
 
-    println!("{}", base.ui.apply(GREY.apply_to(">>> Logged out")));
+    println!("{}", base.ui.apply(GREY.apply_to(output)));
```

Comment:
Minor nit but we now have `cprintln!`in `turborepo-ui` that makes this a little cleaner

Reply:
Adoption of `cprintln` is separate from this; we can go back and make that change throughout the codebase all at once. (It's a mechanical change, unlike, say `turbopath` changes.)